### ISSUE TITLE
Search toolbar

### ISSUE THREAD

**Issue by 75andybermond** [illustrated-machine-learning/illustrated-machine-learning.github.io#9]
It will be great to have a search bar available on the site. 
Additional feature like this will allow any users to request for specific contents. 
- [ ] If developers are interested to work on this please let me know. I would like to help on this.

**Comment by francescodisalvo05** [comment]
It would be an awesome feature. At the moment @bozzelliandrea  is working on the complete porting to a web app in react. Once it is completed, we may be ready to start working on that.

Of course, any sort of collaboration may be useful! 

Thanks for your suggestion :D

**Comment by 75andybermond** [comment]
Happy to help once the task you mentioned is "Done" 😉

**Comment by francescodisalvo05** [comment]
Oh, whait, what do you mean by search bar? 🤔 
Do you wanna look for some _external_ contents or to _switch_ between pages? Because ChatGPT has no access to internet, therefore I don't think it can look for "our" contents between the pages.

**Comment by 75andybermond** [comment]
As a search bar i'd like to access content that are related to my query like "search : unsupervised", then i get a selection of all pages that talk about this subject. 
Hope it is more clear.

**Comment by bozzelliandrea** [comment]
Good point @75andybermond, it's a nice idea, we are evaluating a solution based on [LyraSearch](https://github.com/LyraSearch/lyra) in the future we plan to introduce a search based at least on navigation keywords

**Comment by 75andybermond** [comment]
Great idea!
